adr/RKE-KDM-decouple.md

```diff
@@ -0,0 +1,185 @@
+   "\u003e=1.27.0-rancher1-1": "calico-v3.26.1",
+   "\u003e=1.8.0-rancher0 \u003c1.13.0-rancher0": "calico-v1.8"
+  },
+```
+It's a bit cryptic but each line is stating a min and max Kubernetes release. RKE1 checks the Kubernetes release we are using to know what calico manifest to install. For example, for `v1.26.9-rancher1-1`, we will use the manifest `calico-v3.25.0`.
+
+Part 2 includes the actual manifest, which is also part of `data/data.json`. I will not show it here because it is too long but you can expect a long string representing all required manifests after the variable `calico-v3.25.0=`
+
+### How does RKE1 consume KDM?
+
+As stated in the [README.md](https://github.com/rancher/rke#building), before building RKE1 we must execute `go generate`, which will fetch `data/data.json` from a default URL set by the [defaultURL variable](https://github.com/rancher/rke/blob/release/v1.5/codegen/codegen.go#L13). We can modify that URL with env variables when executing `go generate` to embed a different `data/data.json` into the rke binary.
+
+Note that it is also possible to override the URL at runtime with an environment variable.
+
+### How is the release process right now
+
+It all starts with KDM creating a branch with the name `dev-vX.Y-YEAR-MONTH-patches`, for example [dev-v2.7-2023-11-patches](https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.7-2023-11-patches).
+
+Then, a PR is created updating that KDM branch:
+* The new images that will be used (both Kubernetes and addons)
+* The new manifests if needed (e.g. if there is a new addon version that changes the manifest)
+* The logic to pick up manifests if needed (e.g. if there is a new addon manifest)
+
+Here is [an example](https://github.com/rancher/kontainer-driver-metadata/pull/1257/files)
+
+When merged in KDM, the default URL in RKE1 is changed, pointing at the previously described branch. Here is [an example](https://github.com/rancher/rke/commit/ca8b578f55ffd46c5ff9485a4fcedcb9107f9425).
+
+Once the default URL PR is merged, RKE1 releases an RC. This RC is used by QA (infracloud team) to run its RKE1/KDM tests. If a regression is found, KDM or RKE1 code is updated and a new RC is released. Once QA signs off an RC, that RC can be considered for an "official" RKE1 release. However, we release RKE1 until KDM does an final release. KDM is consumed by other projects, such as Rancher Manager or RKE2, so this final KDM release can be delayed.
+
+When KDM releases, it merges the changes of `dev-vX.Y-YEAR-MONTH-patches` into `release-vX.Y` and the default URL in RKE1 is changed again. Here is [an example](https://github.com/rancher/rke/commit/6e8ec25f09f2ff88f66b0fe643f9d40b84f43756). After that, RKE1 is officially released.
+
+Note that, as of today, RKE1 does not intend to quickly release upstream Kubernetes releases. By the time the KDM branch is created, it is normal that upstream Kubernetes releases are already out.
```

Comment:
> Note that, as of today, RKE1 does not intend to quickly release upstream Kubernetes releases 

This line sounds a bit off. We can probably just say KDM branches are created after upstream Kubernetes releases are out.